libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -277,4 +278,29 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  private boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      return true;
+    } else {
+      permissionBackoff();
+      return false;
+    }
+  }
+
+  private void permissionBackoff() {
+    Handler handler = new Handler();
+    ExponentialBackoff counter = new ExponentialBackoff();
+
+    PermissionCheckRunnable permissionCheckRunnable = new PermissionCheckRunnable(context, this);
+
+    long nextWaitTime = counter.nextBackOffMillis();
+    handler.postDelayed(permissionCheckRunnable, nextWaitTime);
+  }
+
+  void locationAndReceiverSetup() {
+    registerEventReceiver();
+    isOpted = true;
```

Comment:
https://github.com/mapbox/mapbox-events-android/pull/12#discussion_r154335065
> Besides that, `isOpted` should only be `true` if previous actions `startLocation()` and `registerEventReceiver()` were executed, meaning that location is opted.

`startLocation()` and `registerEventReceiver()` might fail, order matters.